src/compiler/transformers/esDecorators.ts

```diff
@@ -2332,4 +2340,59 @@ export function transformESDecorators(context: TransformationContext): (x: Sourc
             factory.createStringLiteral("default") :
             factory.createStringLiteralFromNode(name);
     }
+
+    function createMetadata(name: Identifier, hasProto: boolean) {
+        const varDecl = factory.createVariableDeclaration(
+            name,
+            /*exclamationToken*/ undefined,
+            /*type*/ undefined,
+            factory.createConditionalExpression(
+                factory.createBinaryExpression(
+                    factory.createBinaryExpression(factory.createTypeOfExpression(factory.createIdentifier("Symbol")), factory.createToken(SyntaxKind.EqualsEqualsEqualsToken), factory.createStringLiteral("function")),
+                    factory.createToken(SyntaxKind.AmpersandAmpersandToken),
+                    factory.createPropertyAccessExpression(factory.createIdentifier("Symbol"), factory.createIdentifier("metadata")),
+                ),
+                factory.createToken(SyntaxKind.QuestionToken),
+                factory.createCallExpression(
+                    factory.createPropertyAccessExpression(factory.createIdentifier("Object"), factory.createIdentifier("create")),

```

Comment:
`factory.createPropertyAccesExpression` will create the identifier for you.
```suggestion
                    factory.createPropertyAccessExpression(factory.createIdentifier("Object"), "create"),
```